src/Compilers/VisualBasic/Portable/Symbols/SubstitutedTypeParameterSymbol.vb

```diff
@@ -39,6 +39,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
 
         Public Sub New(originalDefinition As TypeParameterSymbol)
             Debug.Assert(originalDefinition.IsDefinition)
+            Debug.Assert(TypeOf originalDefinition Is SubstitutableTypeParameterSymbol) ' Required to ensure symmetrical equality
```

Comment:
> Why don't we just make this the parameter type here?

I don't think this is necessary and it would complicate consumers that create instances of this type.

---
In reply to: [533606042](https://github.com/dotnet/roslyn/pull/49569#discussion_r533606042) [](ancestors = 533606042)